components/802.15.4_RF/stm-s2lp-rf-driver/source/NanostackRfPhys2lp.cpp

```diff
@@ -695,7 +744,7 @@ static int8_t rf_extension(phy_extension_type_e extension_type, uint8_t *data_pt
             }
             break;
         case PHY_EXTENSION_SET_TX_POWER:
-            // TODO: Set TX output power
+            rf_set_tx_power(*data_ptr);
```

Comment:
It is safe to write the PA_POWER registers only if rf_state == RF_IDLE and when S2-LP is set to S2LP_STATE_READY.
Store the new TX power value here and write it to RF register when rf_receive is called. Use some global "bool rf_update_tx_power" For example, see handling PHY_EXTENSION_SET_CCA_THRESHOLD below.

Reply:
Fixed